test/failed-directories.test.ts

```diff
@@ -1,6 +1,13 @@
 import fs from 'fs';
 import { TempDir } from './util';
-import { cosmiconfig, Options } from '../src';
+import { cosmiconfig as cosmiconfigActual, Options } from '../src';
+
+const cosmiconfig: typeof cosmiconfigActual = (...params) =>
+  require('../src').cosmiconfig(...params);
+
+cosmiconfig.loadJs = cosmiconfigActual.loadJs;
+cosmiconfig.loadJson = cosmiconfigActual.loadJson;
+cosmiconfig.loadYaml = cosmiconfigActual.loadYaml;
```

Comment:
Can you say more about the value added by these dynamic require changes?

Reply:
Using dynamic require is the only way I know how to get jest mocks/spys to correctly work with `util.promisify` (among other issues). Without the change the tests fail.

https://github.com/jest-community/eslint-plugin-jest/issues/283

Reply:
Were you facing any problems in this project, that needed to be remedied? If not, I'd prefer not to add the complexity of this commit.

Reply:
Are you referring to the complexity of adding:

```js
cosmiconfig.loadJs = cosmiconfigActual.loadJs;
cosmiconfig.loadJson = cosmiconfigActual.loadJson;
cosmiconfig.loadYaml = cosmiconfigActual.loadYaml;
```
or
`import { cosmiconfig as cosmiconfigActual, Options } from '../src';`
or 
`const cosmiconfig: typeof cosmiconfigActual = (...params) =>
  require('../src').cosmiconfig(...params);`

Or all three? I ask because I'm planning on putting a PR that exports the loaders as a named export `defaultLoaders` will remove the first example.

In the meantime I'll try and find an alternate solution. I think the issue comes from using `util.promisify` so I can just manually make `fs.readFile` a promise again.

Reply:
I reverted the commit and changed:

```js
import { promisify } from 'util';

const fsReadFileAsync = promisify(fs.readFile);
```

to: 

```js
async function fsReadFileAsync(
  pathname: string,
  encoding: string,
): Promise<string> {
  return new Promise((resolve, reject): void => {
    fs.readFile(pathname, encoding, (error, contents): void => {
      if (error) {
        reject(error);
        return;
      }

      resolve(contents);
    });
  });
}
```